examples/tutorial_2_customproto.md

```diff
@@ -22,29 +22,35 @@ type TestProto = ref object of LPProtocol
 
 We've set a [protocol ID](https://docs.libp2p.io/concepts/protocols/#protocol-ids), and created a custom `LPProtocol`. In a more complex protocol, we could use this structure to store interesting variables.
 
-A protocol generally has two part: and handling/server part, and a dialing/client part.  
-Theses two parts can be identical, but in our trivial protocol, the server will wait for a message from the client, and the client will send a message, so we have to handle the two cases separately.
+A protocol generally has two parts: a handling/server part, and a dialing/client part. These two parts can be identical, but in our trivial protocol, the server will wait for a message from the client, and the client will send a message, so we have to handle the two cases separately.
 
 Let's start with the server part:
 ```nim
 proc new(T: typedesc[TestProto]): T =
-  # every incoming connections will in be handled in this closure
-  proc handle(conn: Connection, proto: string) {.async, gcsafe.} =
-    echo "Got from remote - ", string.fromBytes(await conn.readLp(1024))
-    # We must close the connections ourselves when we're done with it
-    await conn.close()
+  # every incoming connections will be handled in this closure
+  proc handle(stream: Connection, proto: string) {.async, gcsafe.} =
+    echo "[server] received from client: ", string.fromBytes(await stream.readLp(1024))
+    await stream.writeLp("Hi back!")
```

Comment:
It's probably a good idea to wrap it in a `try/except/finally`, to acquaint the reader with best practices, etc...